package.json

```diff
@@ -148,7 +148,7 @@
     "react-native-plaid-link-sdk": "10.8.0",
     "react-native-qrcode-svg": "^6.2.0",
     "react-native-quick-sqlite": "^8.0.0-beta.2",
-    "react-native-reanimated": "3.5.4",
+    "react-native-reanimated": "^3.6.1",
```

Comment:
Getting crash on android on latest main

<img width="337" alt="android crash" src="https://github.com/Expensify/App/assets/108292595/0116263a-8994-4ec5-b8e5-e0b2d072c807">

Reply:
weird. did you try npm ci? and rebuilding Android

Reply:
I removed node_modules, installed npm and did fresh build

Reply:
This also might be related

<img width="337" alt="another" src="https://github.com/Expensify/App/assets/108292595/efc6029c-3abd-410e-b20b-7dc6ee9225bc">

Reply:
@situchan Does `npm start -- --reset-cache` help?

Reply:
Already did everything. I had to revert changes on this PR locally to unblock testing other PRs

Reply:
You're right, I can indeed reproduce this issue on 912fa3fbb145ab903284f598e0a1c882ea819ff5 as well. Working on it!

Reply:
@tomekzaw if possible could you please help us understand why the error comes from `fill`?
Is the API being used incorrectly?




https://github.com/Expensify/App/pull/26654/files#diff-58517876b1c001c5b56c9c446d55dd75c290c4e07c766fdbcb32e0d90d0d42f7R26

Reply:
@rushatgabhane I'm back with answers. Fortunately, it's not a regression in Reanimated 3.6.1 because on 3.5.4 it behaves the same way. It's also not an issue with react-native-svg (13.14.0 is okay).

If you animate `fill` prop of SVG elements, you need to pass `adapter` to `useAnimatedProps`, see the docs:
https://docs.swmansion.com/react-native-reanimated/docs/core/useAnimatedProps/#adapters-

You need the following adapter:
```tsx
  const adapter = createAnimatedPropAdapter(
    (props) => {
      // delete props.fill;
      if (Object.keys(props).includes('fill')) {
        props.fill = { type: 0, payload: processColor(props.fill) };
      }
    },
    ['fill']
  );
```

Make sure to import `processColor` from `react-native-reanimated` (not `react-native`). Then, you pass the adapter as third argument to `useAnimatedProps`:

```tsx
  const animatedProps = useAnimatedProps(
    () => {
      return {
        fill: interpolateColor(animatedValue.value, [0, 1], ['red', 'blue']),
      };
    },
    [],
    adapter
  );
```

Here's the full example for the context:

<details>

```tsx
import React, { useEffect } from 'react';
import Animated, {
  Easing,
  createAnimatedPropAdapter,
  interpolateColor,
  useAnimatedProps,
  useSharedValue,
  processColor,
  withTiming,
} from 'react-native-reanimated';
import Svg, { Path } from 'react-native-svg';

const AnimatedPath = Animated.createAnimatedComponent(Path);

function FabPlusIcon() {
  const animatedValue = useSharedValue(0);

  useEffect(() => {
    animatedValue.value = withTiming(1, {
      duration: 340,
      easing: Easing.inOut(Easing.ease),
    });
  }, [animatedValue]);

  const adapter = createAnimatedPropAdapter(
    (props) => {
      // delete props.fill;
      if (Object.keys(props).includes('fill')) {
        props.fill = { type: 0, payload: processColor(props.fill) };
      }
    },
    ['fill']
  );

  const animatedProps = useAnimatedProps(
    () => {
      return {
        fill: interpolateColor(animatedValue.value, [0, 1], ['red', 'blue']),
      };
    },
    [],
    adapter
  );

  return (
    <Svg width={20} height={20}>
      <AnimatedPath
        d="M12,3c0-1.1-0.9-2-2-2C8.9,1,8,1.9,8,3v5H3c-1.1,0-2,0.9-2,2c0,1.1,0.9,2,2,2h5v5c0,1.1,0.9,2,2,2c1.1,0,2-0.9,2-2v-5h5c1.1,0,2-0.9,2-2c0-1.1-0.9-2-2-2h-5V3z"
        animatedProps={animatedProps}
      />
    </Svg>
  );
}

FabPlusIcon.displayName = 'FabPlusIcon';

export default FabPlusIcon;
```

</details>

Reply:
Yeah, I think we should also add `stroke` there so it looks something like this:
```
  const adapter = createAnimatedPropAdapter(
    (props) => {
      if (Object.keys(props).includes('fill')) {
        props.fill = { type: 0, payload: processColor(props.fill) };
      }
      if (Object.keys(props).includes('stroke')) {
        props.stroke = {type: 0, payload: processColor(props.stroke)}
      }
    },
    ['fill', 'stroke']
  );

```

`useAnimatedSVGProps` sounds good :rocket: